app/services/api/v3/delivery_partners/show.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Api
+  module V3
+    module DeliveryPartners
+      class Show
+        attr_reader :lead_provider, :params
+
+        def initialize(lead_provider:, params:)
+          @lead_provider = lead_provider
+          @params = params
+        end
+
+        def delivery_partner
+          lead_provider.delivery_partners.find_by("delivery_partners.id = ?", params[:id])
```

Comment:
```suggestion
          lead_provider.delivery_partners.find(params[:id])
```
I think if an ID does not exist, or we don't have access to view that delivery partner, we should return a 404 in the API, currently the response exposes the following error instead `PG::InvalidTextRepresentation` and a bad request.